### PR TITLE
✨ Feature ➾ added ability to deploy to mainnet and interact with contracts deployed there

### LINKED ISSUE #1598: 🚧 Dev ➾ Support deployment to mainnet
### 🔗 Epic|Feature Link

_No response_

### 🚥 Definition of Done

- [x] 🎯 Planning and Design
- [ ] 🏎️ Dev Implementation Complete
- [ ] ♾️ PR Merged
- [ ] 🏆 Product Acceptance

### 🧰 Work Description

_No response_

### ✅ To Do

- [ ] Implement support using secret key stored in a hidden file or env variable

### ⚖️ Acceptance Criteria

✨ Functional: 
- [ ] Users can target mainnet for origination and interact with it via transfer
- [ ] Means provided to add a secret key (env variable or separate file or new task to provide)
- [ ] Environment for mainnet 
- [ ] Documentation regarding safety / not checking in to git

### 🔭 Scope

🙅‍♂️ Out of Scope: 
- Hardware signer
 
 

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by jchenche**
# 🌮 Taqueria PR

Fixes #1598

## 🪁 Description

Added the ability to deploy and interact with mainnet.
- Allow mnemonic to be specified in the config.
- Create a new environment called production that points to a newly created network called mainnet.

## 🪂 Pre-Merge Checklist (Definition of Done)

🚦 Required to merge:

- [x] ⛱️ I have completed this PR template in full and updated the title appropriately
- [x] ⛵ My code builds cleanly, and I have manually tested the changes
- [ ] 🏄‍♂️ Another team member has built this branch and done manual testing on the change
- [x] 🏖️ New and existing unit tests pass locally and in CI
- [x] 🐬 I have commented my code, particularly in hard-to-understand areas
- [x] 🤿 Corresponding changes have been made to all documentation
- [x] 🚢 The release checklist has been completed

## 🛸 Type of Change

- [ ] 🧽 Chore ➾
- [ ] 🛠️ Fix ➾
- [x] ✨ Feature ➾
- [ ] 👷 Refactor ➾
- [ ] 🧪 Pre-Release ➾
- [ ] 🚀 Release ➾

### CODE REVIEW

File: taqueria-config.ts [hunk 1]
```diff
@@ -51,6 +55,11 @@ export const defaultConfig: Config.t = Config.create({
 			rpcUrl: 'https://ghostnet.ecadinfra.com',
 			protocol: 'PtKathmankSpLLDALzWw7CGD2j2MtyveTwboEYokqUCP4a1LxMg',
 		},
+		mainnet: {
+			label: 'mainnet',
+			rpcUrl: 'https://mainnet.api.tez.ie',
+			protocol: 'PtKathmankSpLLDALzWw7CGD2j2MtyveTwboEYokqUCP4a1LxMg',
```
Comment (jchenche) [inline taqueria-config.ts:61]: @mweichert just want to confirm that protocol here can be removed right? I recall that it is ignored anyways, for non-sandbox networks.

**Review by hu3man — APPROVED** [review]

File: tests/e2e/data/taquito-funding-instructions.txt [hunk 2]
```diff
@@ -1,4 +1,4 @@
-A keypair with public key hash tz3__address__ was generated for you.
+A keypair with public key hash tz__address__ was generated for you.
 To fund this account:
 1. Go to https://teztnets.xyz and click "Faucet" of the target testnet
```
Comment (egarson) [inline tests/e2e/data/taquito-funding-instructions.txt:3]: Faucets have been removed, no?

File: website/docs/config/networks.mdx [hunk 3]
```diff
@@ -71,3 +67,38 @@ To target the `ghostnet` network we configured above using the `testing` environ
 ```shell
 taq originate <contractName> --env testing
 ```
+
+## Deploying and Transferring with Mainnet
+
+Configuring a mainnet is quite similar to configuring a teztnet with one difference. Users must also create an account, with the mnemonic specified, within the network in order to interact with mainnet. See the example below:
+
+:::danger
+Improvements are on the way. However, we currently only support signing in memory and we require a mnemonic to be written in the config file, so do not commit this file or delete the mnemonic before doing so). This is not the safest way to interact with mainnet so proceed with caution
```
Comment (egarson) [inline website/docs/config/networks.mdx:76]: Missing opening bracket

**Review by egarson — APPROVED** [review]
LGTM with tiny changes.
Nice work @jchenche

File: tests/e2e/data/taquito-funding-instructions.txt (hunk 2 above)
Comment (jchenche) [reply tests/e2e/data/taquito-funding-instructions.txt:3]: Yep but the meaning of Faucet here is related to that website

File: website/docs/config/networks.mdx (hunk 3 above)
Comment (jchenche) [reply website/docs/config/networks.mdx:76]: Will fix!

Comment (jchenche) [reply website/docs/config/networks.mdx:76]: Forgot to delete closing bracket actually

**Pull request merged**